pkg/storegateway/bucket_chunk_reader.go

```diff
@@ -211,10 +209,11 @@ func (r *bucketChunkReader) loadChunks(ctx context.Context, res []seriesEntry, s
 	return nil
 }
 
-// save saves a copy of b's payload to a memory pool of its own and returns a new byte slice referencing said copy.
-// Returned slice becomes invalid once r.block.chunkPool.Put() is called.
-func (r *bucketChunkReader) save(b []byte) ([]byte, error) {
-	dst, err := r.chunkBytes.Get(len(b))
+// save a copy of b's payload to a buffer pulled from chunksPool.
+// The buffer containing the chunk data is returned.
+// The returned slice becomes invalid once chunksPool is released.
+func (r *bucketChunkReader) save(b []byte, chunksPool *pool.BatchBytes) ([]byte, error) {
```

Comment:
This doesn't have to be a method of `bucketChunkReader` anymore

Reply:
And even better - what do you think about inlining it `populateChunk`? It's simple enough and it's not used elsewhere

Reply:
Makes sense to me.

Reply:
I will also move `populateChunk()` to `bucket_chunk_reader.go`.

Reply:
While doing it, I realized that inlining `save()` logic input `populateChunk()` makes the code harder to read. I will keep other suggestions tho.

Reply:
See: https://github.com/grafana/mimir/pull/3611/commits/c2a2039700474511d89fff28b8cf71b261ea36bb

Reply:
yeah I'm happy with what you have. also removing the indirection of passing a func is good enough

Reply:
Thanks!